### PR TITLE
fix(vite-node): have a separate cache for web/ssr transforms

### PR DESCRIPTION
**Pull request by sheremet-va**
### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [ ] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Comment by sheremet-va** [comment]
@AriPerkkio this should help you with the coverage, right?

File: packages/vite-node/src/server.ts [hunk 1]
```diff
@@ -11,19 +11,33 @@ import { withInlineSourcemap } from './source-map'
...
-    duration?: number
-    timestamp: number
-    result: FetchResult
-  }>()
+  fetchCaches = {
+    ssr: new Map<string, FetchCache>(),
+    web: new Map<string, FetchCache>(),
+  }
+
+  fetchCache = new Map<string, FetchCache>()
```
Comment (AriPerkkio) [inline packages/vite-node/src/server.ts:40]: Is this one left here for backwards compatibility or why is it needed?

**Review by AriPerkkio — COMMENTED** [review]
Yes this is needed for https://github.com/vitest-dev/vitest/issues/3888, thank you!

We need to remember to make runners pass transformMode to coverage providers before releasing v1. Just so that it would not cause a breaking change. The fix for #3888 can then be implemented later.

File: packages/vite-node/src/server.ts (hunk 1 above)
Comment (sheremet-va) [reply packages/vite-node/src/server.ts:40]: Yes

**Pull request merged**